src/JustEat.StatsD/StatsDPublisher.cs

```diff
@@ -127,11 +131,31 @@ public void MarkEvent(string name)
             Send(_formatter.Event(name));
         }
 
+
+
         private void Send(string metric)
         {
+            if (string.IsNullOrEmpty(metric))
+            {
+                return;
+            }
+
             try
             {
-                _transport.Send(metric);
+#if NETCOREAPP2_1
+                if (metric.Length <= 128)
+                {
+                    Span<byte> destination = stackalloc byte[metric.Length * 2];
+                    var written = Encoding.UTF8.GetBytes(metric, destination);
```

Comment:
If we're optimising, would pulling the value of `Encoding.UTF8` into a static field as a one-time operation and then just using that speed things up?